src/makeInfinityAdder.js

```diff
@@ -5,6 +5,23 @@
  */
 function makeInfinityAdder() {
   // write code here
+  let sum = 0;
+
+  const adder = (number) => {
+    if (number === undefined) {
+      const currentSum = sum;
+
+      sum = 0;
+
+      return currentSum;
+    }
+
+    sum += number;
+
+    return adder;
+  };
+
+  return adder;
```

Comment:
The `makeInfinityAdder` function correctly creates a closure over `sum` and returns the `adder` function. However, the `adder` function should not accept any arguments other than a number or `undefined`. Currently, it accepts any value, which might lead to unexpected behavior if non-numeric values are passed. To ensure the function adheres strictly to the task requirements, you may consider adding a check to ensure that the argument is either a number or `undefined`.